plugins/inputs/http/http.go

```diff
@@ -116,13 +136,69 @@ func (h *HTTP) Init() error {
 
 	h.client = client
 
+	if h.CookieAuthURL != "" {
+		// cookie auth defaults
+		renewalInterval := defaultCookieAuthRenewal
+		if h.CookieAuthRenewal != 0 {
+			renewalInterval = time.Duration(h.CookieAuthRenewal)
+		}
+		if h.CookieAuthMethod == "" {
+			h.CookieAuthMethod = defaultCookieAuthMethod
+		}
+		// add cookie jar to HTTP client
+		if h.client.Jar, err = cookiejar.New(nil); err != nil {
+			return err
+		}
+		// start auth ticker
+		if authErr := h.startCookieAuth(renewalInterval); authErr != nil {
+			return authErr
+		}
+	}
+
 	// Set default as [200]
 	if len(h.SuccessStatusCodes) == 0 {
 		h.SuccessStatusCodes = []int{200}
 	}
 	return nil
 }
 
+func (h *HTTP) startCookieAuth(interval time.Duration) error {
+	// continual auth ticker
+	ticker := time.NewTicker(interval)
+	go func() {
+		for range ticker.C {
+			_ = h.doCookieAuth()
+		}
+	}()
+
+	// initial auth will immediately error out the Init() if auth fails
+	return h.doCookieAuth()
+}
+
+func (h *HTTP) doCookieAuth() error {
+	request, err := h.newRequest(h.CookieAuthURL, h.CookieAuthMethod, h.CookieAuthBody)
+	if err != nil {
+		return err
+	}
+	defer request.Body.Close()
+
+	errBadRespCode := fmt.Errorf("bad response code")
+	resp, respErr := h.client.Do(request)
+	if respErr != nil {
+		return respErr
+	}
+	if _, err = io.Copy(ioutil.Discard, resp.Body); err != nil {
+		return err
+	}
+	if err = resp.Body.Close(); err != nil {
+		return err
+	}
```

Comment:
I think you want this to be
```suggestion
	resp, respErr := h.client.Do(request)
	if respErr != nil {
		return respErr
	}
	defer resp.Body.Close()

	if _, err := io.Copy(ioutil.Discard, resp.Body); err != nil {
		return err
	}
```
to avoid skipping the body-close if the copy fails (which should never be the case I guess).

Reply:
Sure.  Force of habit to check errors.